service/history/workflow/context.go

```diff
@@ -894,10 +880,14 @@ func (c *ContextImpl) maxHistorySizeExceeded(shardContext shard.Context) bool {
 			tag.WorkflowRunID(c.workflowKey.RunID),
 			tag.WorkflowHistorySize(historySize),
 			tag.WorkflowEventCount(historyCount))
-
-		return true
+		if err := c.forceTerminateWorkflow(ctx, shardContext, fmt.Sprintf(
```

Comment:
I have a minor performance concern about using `fmt.Sprintf`. Previously I have been advised against using it on hot code paths, but maybe it is okay here.

Reply:
fmt.Sprintf is no longer needed.